spv/sync.go

```diff
@@ -1155,110 +1286,135 @@ func (s *Syncer) getHeaders(ctx context.Context, rp *p2p.RemotePeer) error {
 	}
 }
 
+func (s *Syncer) fetchMissingCFilters(ctx context.Context, rp *p2p.RemotePeer) error {
+	for walletID, w := range s.wallets {
+		s.fetchMissingCfiltersStart(walletID)
+		progress := make(chan wallet.MissingCFilterProgress, 1)
+		go w.FetchMissingCFiltersWithProgress(ctx, rp, progress)
+
+		for p := range progress {
+			if p.Err != nil {
+				return p.Err
+			}
+			s.fetchMissingCfiltersProgress(walletID, p.BlockHeightStart, p.BlockHeightEnd)
+		}
+		s.fetchMissingCfiltersFinished(walletID)
+	}
+	return nil
+}
+
 func (s *Syncer) startupSync(ctx context.Context, rp *p2p.RemotePeer) error {
+	log.Infof("Syncing %d wallets", len(s.wallets))
+
+	_, _, highestChainWallet := s.highestChainTip(ctx)
+
 	// Disconnect from the peer if their advertised block height is
 	// significantly behind the wallet's.
-	_, tipHeight := s.wallet.MainChainTip(ctx)
+	_, tipHeight := highestChainWallet.MainChainTip(ctx)
 	if rp.InitialHeight() < tipHeight-6 {
 		return errors.E("peer is not synced")
 	}
-	s.fetchMissingCfiltersStart()
-	progress := make(chan wallet.MissingCFilterProgress, 1)
-	go s.wallet.FetchMissingCFiltersWithProgress(ctx, rp, progress)
 
-	for p := range progress {
-		if p.Err != nil {
-			return p.Err
-		}
-		s.fetchMissingCfiltersProgress(p.BlockHeightStart, p.BlockHeightEnd)
-	}
-	s.fetchMissingCfiltersFinished()
+	s.fetchMissingCFilters(ctx, rp)
```

Comment:
`s.fetchMissingCFilters` returns an error that should not be ignored. If the error returned is not nil, it should be returned here.
```suggestion
	if err := s.fetchMissingCFilters(ctx, rp); err != nil {
		return err
	}
```